pkg/operation/botanist/controlplane/resourcemanager/resource_manager.go

```diff
@@ -0,0 +1,553 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcemanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/utils"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/secrets"
+
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	autoscalingv1beta2 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1beta2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// SecretName is a constant for the secret name for the gardener resource manager's kubeconfig secret.
+	SecretName = "gardener-resource-manager"
+	// UserName is the name that should be used for the secret that the gardener resource manager uses to
+	// authenticate itself with the kube-apiserver (e.g., the common name in its client certificate).
+	UserName = "gardener-resource-manager"
```

Comment:
```suggestion
	UserName = "gardener.cloud:system:gardener-resource-manager"
```

I know, we spoke about it, but for now let's keep this old name as part of this refactoring PR. If needed, we can change it later/separately.

Reply:
Please, keep the name as it is before the PR. There are landscapes where these user names are used to define the audit policies. Just the leader election requests from the controller is generating too much unwanted audit events that might be already filtered out with a policy.

Reply:
I am confused. I did not change the name in this PR as discussed. See: https://github.com/gardener/gardener/pull/3445/files#diff-756b94b9db03912f92e00c131c6e8632a4f8c2ed8f3a3b8f49445919c639ac2aL28

Reply:
@BeckerMax what you reference is the name of the service account. However, you use the `UserName` constant [here](https://github.com/gardener/gardener/pull/3445/files#diff-be4a114eed4ba1e885b1e46ba2489834b18b3e0471575f78948d40588ad8dceaL293) and this is clearly a change. Please revert it.